expand.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"bytes"
+	"golang.org/x/text/transform"
+	"io"
+)
+
+// expandReadCloser implements io.ReadCloser
+type expandReadCloser struct {
+	tReader *transform.Reader
+	closer  io.Closer
+}
+
+func newExpandReadCloser(rc io.ReadCloser, expandFunc func(string) (string, error)) *expandReadCloser {
+	return &expandReadCloser{
+		tReader: transform.NewReader(
+			rc,
+			&expandTransformer{expand: expandFunc}),
+		closer: rc,
+	}
+}
+
+func (this *expandReadCloser) Read(p []byte) (int, error) {
```

Comment:
I don't really care about the rule, some linters are using all the rules from the wiki to validate code(e.g. my VSCode) so I just want to keep them happy. Lets use `e` for the `expandReadCloser` struct.